packages/office-ui-fabric-react/src/components/Callout/CalloutContent.tsx

```diff
@@ -76,7 +76,8 @@ export class CalloutContent extends BaseComponent<ICalloutProps, ICalloutState>
   }
 
   public componentWillUpdate(newProps: ICalloutProps) {
-    if (newProps.targetElement !== this.props.targetElement || newProps.target !== this.props.target) {
+    // If the target element changed, find the new one. If we are tracking target with class name, always find element because we do not know if fabric has rendered a new element and disposed the old element.
+    if (newProps.targetElement !== this.props.targetElement || newProps.targetElement == null) {
```

Comment:
When target is a string for the Element's className, we want to update the location since react may dispose and render a new version, even is the class name is still the same.